src/realm/array.cpp

```diff
@@ -1971,6 +1971,106 @@ std::pair<ref_type, size_t> Array::get_to_dot_parent(size_t ndx_in_parent) const
     return std::make_pair(get_ref(), ndx_in_parent);
 }
 
+
```

Comment:
I see your point. It wasn't entirely obvious from the outset though. I just chose to move the code upwards, rather than downwards, and that ended up generating a less clear diff. Sorry.